Sources/MapboxDirections/RouteRefreshResponse.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+public struct RouteRefreshResponse {
+    public let httpResponse: HTTPURLResponse?
+    
+    public let identifier: String?
+    public var routeIndex: Int
+    public var legIndex: Int
+    
+    public var route: Route? // pass it in a user info? and then consturct new route during decoding?
+    
+    public let credentials: DirectionsCredentials
+    
+    private var legAnnotations: [RouteLegAnnotation]
+    
+    /**
+     The time when this `RouteRefreshResponse` object was created, which is immediately upon recieving the raw URL response.
+     
+     If you manually start fetching a task returned by `Directions.url(forCalculating:)`, this property is set to `nil`; use the `URLSessionTaskTransactionMetrics.responseEndDate` property instead. This property may also be set to `nil` if you create this result from a JSON object or encoded object.
+     
+     This property does not persist after encoding and decoding.
+     */
+    public var created: Date = Date()
+}
+
+extension RouteRefreshResponse: Codable {
+    enum CodingKeys: String, CodingKey {
+        case identifier = "uuid"
+        case route
+    }
+    
+    private func applyRefreshingTo(route: Route) -> Route{
+        let updatedLegs = zip(legAnnotations, route.legs.enumerated()).map { (pair) -> RouteLeg in
+            let (annotation, legEnum) = pair
+            let leg = legEnum.element.copy() as! RouteLeg
+            if legEnum.offset >= legIndex {
+                leg.updateAnnotationData(from: annotation)
+            }
+            return leg
+        }
+        
+        let route = Route(legs: updatedLegs,
+                           shape: route.shape,
+                           distance: route.distance,
+                           expectedTravelTime: route.expectedTravelTime)
+        route.routeIdentifier = route.routeIdentifier
```

Comment:
I think the intention was to copy the original route’s UUID into the new route, but this statement ends up being a no-op because of the shadowed variable.